src/charm.py

```diff
@@ -107,20 +107,28 @@ def _configure(self):
 
         # If the startup arguments are the same and we use the
         # we lifecycle, sent the config reload HTTP request instead
-        if (
-            current_layer
-            and "services" in current_layer
-            and not DeepDiff(current_layer["services"], new_layer["services"])
-        ):
+
+        services_changed = not DeepDiff(
```

Comment:
Doesn't Pebble itself check if there is a change to service layers ? If not shouldn't such a feature at least have an associated issue ticket in Operator Framework of Pebble ?

Reply:
In principle what you say sounds sensible, but what we would be doing it sending to a different container, the one running Pebble, the new service plan over RPC, and use the result of a diff check over there to decide whether to trigger a restart from the charm container. Sounds a bit much to me.

Reply:
@balbirthomas `add_layer` is stateless: https://github.com/canonical/operator/blob/7344b5c284ff4009dbde29c4df99205eaad2e8ba/ops/pebble.py#L870

We could add an API to Pebble to check for diffs, but I am ambivalent on whether it is scope creep for `ops.pebble` or not. @jnsgruk WDYT?